Production/python/get_py.py

```diff
@@ -0,0 +1,162 @@
+                for fileArray in fileArrays:
+                    if fileArray["logical_file_name"] in filelist:
+                        nevents += fileArray["event_count"]
+                nevents_all.append(nevents)
+            
+            # check for sites with 100% dataset presence (based on PhEDEx)
+            # refs:
+            # https://github.com/dmwm/DAS/blob/master/src/python/DAS/services/combined/combined_service.py
+            # https://github.com/gutsche/scripts/blob/master/PhEDEx/checkLocation.py
+            if makese:
+                # get total number of expected files
+                nfiles_tot = len(filelist)
+                # calculate dataset fraction (presence) in % and check for completion
+                highest_percent = 0
+                for site,addr in sitelist:
+                    this_percent = float(sitelist[(site,addr)])/float(nfiles_tot)*100
+                    sitelist[(site,addr)] = this_percent
+                    if this_percent > highest_percent: highest_percent = this_percent
+            
+                sfile.write(f+"\n")
+                if highest_percent < 100:
+                    sfile.write("  !!! No site has complete dataset !!! ( Highest: "+str(highest_percent)+"% )\n")
+                for site,addr in sorted(sitelist):
+                    this_percent = sitelist[(site,addr)]
+                    if this_percent==highest_percent:
+                        sfile.write("  "+site+" ("+addr+")\n")
+    
+            if makepy:
+                #sort list of files for consistency
+                filesort = sorted(filelist)
+                counter = 0
+                #split into chunks of 255
+                for lfn in filesort:
+                    if counter==0: pfile.write("readFiles.extend( [\n")
+                    pfile.write("       '"+lfn+"',\n")
+                    if counter==254 or lfn==filesort[-1]:
+                        pfile.write("] )\n")
+                        counter = 0
+                    else:
+                        counter += 1
+    
+        #only do weightproducer stuff for MC
+        if makewp and is_data==False:
+            nevents = nevents_all[0]
+            neff = 0
+            if len(x)>0: neff = x[0]
+            
+            #handle combining extended samples
+            if len(ff)>1:
+                neff = sum(x[0:])
+                nevents = sum(nevents_all)
+            
+            for i,f in enumerate(ff):
+                #make line for weightproducer
+                line = (" "*8)+repr(MCSample(f.split('/')[1],"-".join(f.split('/')[2].split('-')[1:3]),f.split('/')[2].split('-')[0],"Constant",nevents,neff if neff>0 else None))+","
+                if neff>0:
+                    if len(ff)>1: line = line+" # subtotal = "+str(x[i])+", straight subtotal = "+str(nevents_all[i])+"\n"
+                    else: line = line+" # straight total = "+str(nevents)+"\n"

```

Comment:
doesn't need to print "# straight total = ..." as a comment anymore, because now we store it in the `MCSample` directly